app/services/builders/external_pull_request.rb

```diff
@@ -8,17 +8,27 @@ def initialize(pull_request_event_data, external_project)
     def call
       pull_request_payload = @pull_request_event_data.dig(:payload, :pull_request)
 
-      ::ExternalPullRequest
-        .find_or_create_by!(github_id: pull_request_payload[:id]) do |pull_request|
-        pull_request.html_url = pull_request_payload[:html_url]
-        pull_request.body = pull_request_payload[:body]
-        pull_request.title = pull_request_payload[:title]
-        pull_request.opened_at = pull_request_payload[:created_at]
-        pull_request.owner_id = User.find_by!(
+      pull_request = ::ExternalPullRequest
+                     .find_or_create_by!(github_id: pull_request_payload[:id]) do |ext_pull_request|
+        ext_pull_request.html_url = pull_request_payload[:html_url]
+        ext_pull_request.body = pull_request_payload[:body]
+        ext_pull_request.title = pull_request_payload[:title]
+        ext_pull_request.opened_at = pull_request_payload[:created_at]
+        ext_pull_request.owner_id = User.find_by!(
           login: @pull_request_event_data.dig(:actor, :login)
         ).id
-        pull_request.external_project_id = @external_project.id
+        ext_pull_request.external_project_id = @external_project.id
       end
+      pull_request.update!(state: assign_state(pull_request_payload))
```

Comment:
To avoid accessing the DB twice, we could just do a `find_or_initialize_by` instead of a `find_or_create_by!`

Reply:
In that case you're right, but if the record is not present, we would hit the DB twice, first with the `create` and then with the `update`. If you use `find_or_initialize_by`, the DB will be hit just once.
Or am I missing something?

Reply:
you are right